app/services/computacenter_ledger_csv.rb

```diff
@@ -0,0 +1,101 @@
+require 'csv'
+
+class ComputacenterLedgerCsv
+  def initialize(users: nil)
+    @users = users
+  end
+
+  def to_csv
+    CSV.generate do |csv|
+      csv << headers
+
+      ledger_entries.each do |ledger_entry|
+        csv << [
+          ledger_entry.first_name,
+          ledger_entry.last_name,
+          ledger_entry.email,
+          ledger_entry.telephone,
+          ledger_entry.responsible_body,
+          ledger_entry.responsible_body_urn,
+          ledger_entry.cc_sold_to_number,
+          ledger_entry.school,
+          ledger_entry.school_urn,
+          ledger_entry.cc_ship_to_number,
+          ledger_entry.updated_at_timestamp.utc.strftime('%d/%m/%Y'),
+          ledger_entry.updated_at_timestamp.utc.strftime('%R'),
+          ledger_entry.updated_at_timestamp.utc.iso8601,
+          ledger_entry.type_of_update,
+          ledger_entry.original_email,
+        ]
+      end
+    end
+  end
+
+private
+
+  def computacenter_identifier(responsible_body)
+    return nil if responsible_body.nil?
+
+    case responsible_body.type
+    when 'LocalAuthority'
+      "LEA#{responsible_body.gias_id}"
+    when 'Trust'
+      "t#{responsible_body.companies_house_number.to_i}" # QUESTION: is `to_i` needed? is it to remove padded zeros?
+    end
+  end
+
+  # TODO: will be moved to model
+  def first_name(user)
+    (user.full_name || '').strip.split(' ').first
+  end
+
+  # TODO: will be moved to model
+  def last_name(user)
+    (user.full_name || '').strip.split(' ').last
+  end
+
+  def users
+    @users ||= User.who_can_order_devices.where('privacy_notice_seen_at IS NOT NULL').limit(1)
+  end
+
+  def ledger_entries
+    users.map do |user|
+      ComputacenterUserLedger.new(
+        user_id: user.id,
+        first_name: first_name(user),
+        last_name: last_name(user),
+        email: user.email_address,
+        telephone: user.telephone,
+        responsible_body: user.responsible_body&.name,
```

Comment:
This isn't quite right – the responsible body should always present but is differently derived based on whether user is a school user or an RB users:

`responsible_body: (user.responsible_body || user.school.responsible_body).name`

Reply:
ok this is now a method the user `derived_responsible_body`, not sure if that's the best name

Reply:
we've used `effective_xyz` elsewhere, so this could be `effective_responsible_body`